specifyweb/specify/migrations/0003_cotype_picklist.py

```diff
@@ -0,0 +1,87 @@
+from django.db import migrations
+
+PICKLIST_NAME = 'CollectionObjectType'
+FIELD_NAME = 'collectionObjectType'
+COTYPE_TEXT = 'Collection Object Type'
+
+def create_cotype_picklist(apps):
+    Collection = apps.get_model('specify', 'Collection')
+    Picklist = apps.get_model('specify', 'Picklist')
+    # Create a cotype picklist for each collection
+    for collection in Collection.objects.all():
+        Picklist.objects.get_or_create(
+            name=PICKLIST_NAME,
+            issystem=True,
+            readonly=True,
+            sizelimit=-1,
+            sorttype=1,
+            type=1,
+            tablename='collectionobjecttype',
+            collection=collection,
+            formatter=PICKLIST_NAME
+        )
+
+def revert_cotype_picklist(apps):
+    Picklist = apps.get_model('specify', 'Picklist')
+    Picklist.objects.filter(name=PICKLIST_NAME).delete()
+
+
+def create_cotype_splocalecontaineritem(apps):
+    Splocalecontainer = apps.get_model('specify', 'Splocalecontainer')
+    Splocalecontaineritem = apps.get_model('specify', 'Splocalecontaineritem')
+    # Create a Splocalecontaineritem record for each CollectionObject Splocalecontainer
+    # NOTE: Each discipline has its own CollectionObject Splocalecontainer
+    for container in Splocalecontainer.objects.filter(name='collectionobject', schematype=0):
+        Splocalecontaineritem.objects.get_or_create(
+            name=FIELD_NAME,
+            picklistname=PICKLIST_NAME,
+            type='ManyToOne',
+            container=container,
+            isrequired=True
+        )
+
+def revert_cotype_splocalecontaineritem(apps):
+    Splocalecontaineritem = apps.get_model('specify', 'Splocalecontaineritem')
+    Splocalecontaineritem.objects.filter(name=FIELD_NAME, container__name='collectionobject', container__schematype=0).delete()
+
+
+def create_cotype_splocaleitemstr(apps):
+    Splocalecontaineritem = apps.get_model('specify', 'Splocalecontaineritem')
+    Splocaleitemstr = apps.get_model('specify', 'Splocaleitemstr')
+    # Create caption & description records for collectionObjectType in Schema Config
+    for container_item in Splocalecontaineritem.objects.filter(name=FIELD_NAME):
+        Splocaleitemstr.objects.get_or_create(
+            language='en',
+            text=COTYPE_TEXT,
+            itemname=container_item
+        )
+        Splocaleitemstr.objects.get_or_create(
+            language='en',
+            text=COTYPE_TEXT,
+            itemdesc=container_item
+        )
```

Comment:
(optional)
Same comment as from https://github.com/specify/specify7/pull/5191#discussion_r1717503127! 😃

Although I think it is even safer now then how it was: I don't think it is possible to unintentionally create extra splocaleitemstr objects if everything is configured correctly because we are using the App Registry and the Historical Models, and we're only creating `collectionObjectType` SpLocaleContainerItems in this migration. 

Though it still looks a _little_ weird that the filtering here implies it is creating labels/descriptions for all fields with the name `collectionObjectType` and the reverse is deleting labels/descriptions for SpLocaleContainerItems specifically in the non-workbench CollectionObject container 😅 
(so this change would be completely optional as far as I can tell)